cpp/ql/src/semmle/code/cpp/internal/ResolveClass.qll

```diff
@@ -1,9 +1,9 @@
 import semmle.code.cpp.Type
 
 pragma[noinline]
-private string getTopLevelClassName(@usertype c) {
+private string getTopLevelClassMangledName(@usertype c) {
   isClass(c) and
-  usertypes(c, result, _) and
+  mangled_type_name(c, result) and
   not namespacembrs(_, c) and // not in a namespace
   not member(_, _, c) and // not in some structure
   not class_instantiation(c, _) // not a template instantiation
```

Comment:
These three `not ...` conjuncts can be removed now that we have proper mangled names, right? AFAIR, we added them to avoid confusing `Foo::Bar` with `Bar`. That also means the name of this predicate can change from `getTopLevelClassMangledName` to `getClassMangledName`.

Reply:
This PR deliberately doesn't change behaviour, to make it easier to see what changes/breaks. I'll generalise it in a future PR.